ydb/library/yql/providers/s3/provider/yql_s3_dq_integration.cpp

```diff
@@ -97,43 +107,74 @@ class TS3DqIntegration: public TDqIntegrationBase {
             maxPartitions = std::max(*mbLimitHint / maxTaskRatio, ui64{1});
             YQL_CLOG(TRACE, ProviderS3) << "limited max partitions to " << maxPartitions;
         }
-
-        if (maxPartitions && parts.size() > maxPartitions) {
-            if (const auto extraParts = parts.size() - maxPartitions; extraParts > maxPartitions) {
-                const auto partsPerTask = (parts.size() - 1ULL) / maxPartitions + 1ULL;
-                for (auto it = parts.begin(); parts.end() > it;) {
-                    const auto to = it;
-                    const auto up = to + std::min<std::size_t>(partsPerTask, std::distance(to, parts.end()));
-                    for (auto jt = ++it; jt < up; ++jt)
-                        std::move(jt->begin(), jt->end(), std::back_inserter(*to));
-                    it = parts.erase(it, up);
-                }
-            } else {
-                const auto dropEachPart = maxPartitions / extraParts;
-                for (auto it = parts.begin(); parts.size() > maxPartitions;) {
-                    const auto to = it + dropEachPart;
-                    it = to - 1U;
-                    std::move(to->begin(), to->end(), std::back_inserter(*it));
-                    it = parts.erase(to);
+        NActors::TActorId fileQueueActor;
+        if (useRuntimeListing) {
+            fileQueueActor = NActors::TActivationContext::ActorSystem()->Register(NDq::CreateS3FileQueueActor(
+                0ul,
+                parts.front(),
+                1000 * maxPartitions,
+                State_->Configuration->FileSizeLimit,
+                useRuntimeListing,
+                maxPartitions,
```

Comment:
True, but it is the best assumption in this scope. In case there are less consumers FileQueue will die on timeout. Solution could be improved by getting the actual amount of tasks from executor, but I would suggest doing it in a separate PR, probably as a part of moving FileQueue creation to runtime.